Watchman.DomainModel/Users/Queries/Handlers/GetWarnEventsQueryHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Watchman.Cqrs;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Users.Queries.Handlers
+{
+    public class GetWarnEventsQueryHandler : IQueryHandler<GetWarnEventsQuery, GetWarnEventsQueryResults>
+    {
+        private readonly ISessionFactory _sessionFactory;
+
+        public GetWarnEventsQueryHandler(ISessionFactory sessionFactory)
+        {
+            this._sessionFactory = sessionFactory;
+        }
+
+        public GetWarnEventsQueryResults Handle(GetWarnEventsQuery query)
+        {
+            using var session = this._sessionFactory.Create();
+
+            var warnEvents = session.Get<WarnEvent>().ToList();
+            List<WarnEvent> filteredEvents = new List<WarnEvent>();
+
+            if (query.ServerId == 0)
```

Comment:
w sumie to można to zrobić po prostu dodając where po if 
tam jest lazy loading, na tym etapie jedynie generujesz instrukcje, a nie operujesz na rzeczywistych danych
.
.
no chyba że zrobisz ToList, wtedy juz masz rzeczywiste dane i na to trzeba uważać
czasami potrafi mocno pomóc, czasami mocno przeszkodzić

Reply:
zrobiłem tak, bo miałem jakieś dzikie exception jak chciałem w jednej linijce dać warunek do filtru

Reply:
spójrz jak wygląda to w innych miejscach

Reply:
już jest twoja sugerowana wersja